library/src/main/java/com/android/volley/RequestQueue.java

```diff
@@ -343,6 +356,8 @@ void finish(Request<?> request) {
                 }
             }
         }
+
+        mRequestLogger.log(request);
```

Comment:
Yeah. I think the problem is that we(and volley) are using the request as a stateful object that does many things. This state stuff should probably be stored separately eventually.